common/djangoapps/util/tests/test_submit_feedback.py

```diff
@@ -303,7 +360,85 @@ def test_valid_request_auth_user(self, zendesk_mock_class, datadog_mock):
             )
         ]
         self.assertEqual(zendesk_mock_instance.mock_calls, expected_zendesk_calls)
-        self._assert_datadog_called(datadog_mock, with_tags=False)
+        self._assert_datadog_called(datadog_mock, [])
+
+    @data(
+        ("course-v1:testOrg+testCourseNumber+testCourseRun", True),
+        ("course-v1:testOrg+testCourseNumber+testCourseRun", False),
+        ("", None),
+        (None, None)
+    )
+    @unpack
+    @override_settings(ZENDESK_CUSTOM_FIELDS={"course_id": 1234, "enrollment_mode": 5678})
+    def test_valid_request_auth_user_with_custom_fields(self, course_id, enrollment_state, zendesk_mock, datadog_mock):
```

Comment:
Unfortunately there is a lot of duplication and hardcoding of values in these tests, making them brittle in my opinion and in need of a refactor. I made a couple of attempts at DRYing them up and but found that those changes made the tests less clear and even more brittle, thanks to all the hardcoding. Given that this test file is more than 500 lines long, I didn't want to take on that refactor as part of this work and instead opted to use ddt to keep my tests semi-dry. If you feel strongly about it I will make another attempt at refactoring them.